### ISSUE TITLE
Chore: Bump Gradle to version 8.8

### ISSUE THREAD

**Issue by rbair23** [hashgraph/hedera-services#13753]
Version 8.6 of Gradle prints a warning with the latest 21.0.3 build of Java. We should upgrade to Gradle 8.8.

**Comment by rbair23** [comment]
Ah, I'm not sure this actually removes the warning.

**Comment by jsync-swirlds** [comment]
The warning about 21.0.1 is because the build specifies that java version.  Not sure if it's possible, but specifying only the major version (21) would avoid warnings for every bugfix update.

**Comment by jjohannes** [comment]
The issue should be: **Update JDK to 21.0.3**

We pin down the exact JDK version for reproducible builds. Gradle does not yet support providing an exact version automatically (https://github.com/gradle/gradle/issues/16628). Therefore, everyone running the build (CI or user) need to make sure the version is installed and Gradle starts with it (documentation will also explain that once #12813 is merged).

Right now the version is defined in multiple places in GH Actions and in one place in Gradle. This could be simplified by defining the version in a separate file and using it in GH Actions via the `java-version-file` configuration of the [action/setup-java](https://github.com/actions/setup-java?tab=readme-ov-file#usage). Then there would only be one place. But that's a GH Actions topic. We could create a separate issue for that (cc @rbarkerSL) 

Anyway, I created two PRs for both updates (Java and Gradle). This shows the places that need to touched in this repo. I don't know if the JDK version is also defined somewhere else as well in other CI/CD automation outside this repo:
- https://github.com/hashgraph/hedera-services/pull/13756
- https://github.com/hashgraph/hedera-services/pull/13757

**Comment by jjohannes** [comment]
Gradle has been upgraded to 8.8 by #13757
